### PR TITLE
Update processing set to handle S3 bucket URIs

### PR DESCRIPTION
**Pull request by amcnicho**
Changes to the visibility read and load functions for #140

### CODE REVIEW

**Comment by amcnicho** [comment]
As long as AWS credentials are configured (documented [here](https://s3fs.readthedocs.io/en/latest/#credentials) and [here](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#configuring-credentials)), after the refactoring of #149 the following _already works_:
```python
import s3fs
from xradio._utils.zarr.common import _open_dataset

s3 = s3fs.S3FileSystem(anon=False, requester_pays=False)

for processing_set in s3.listdir("s3://viper-test-data/Antennae_North.cal.lsrk.split.vis.zarr/"):
    for msv4 in s3.listdir(processing_set["name"]):

        # note that prepending the store arg is necessary to avoid `FileNotFoundError`
        # otherwise it's assumed they're stored locally on disk

        test_output = _open_dataset("s3://"+msv4["name"])
```
This because zarr can interpret S3 bucket URLs using its [FSStore](https://zarr.readthedocs.io/en/stable/api/storage.html#zarr.storage.FSStore) wrapper around the general MutableMapping interface exposed by `fsspec`, which should be accepted as a dependency.

**Comment by amcnicho** [comment]
There is a new test added for the case of S3 reads, and that's what failed in the checks on this branch. I thought I'd set up the test dataset bucket as completely public, but clearly that configuration (or the initialization of s3fs) is insufficient to yield a passing test. We'll need to consider how to securely provide AWS credentials to the service account running tests, and then generalize to a coherent policy of support for S3 bucket access.

**Comment by Jan-Willem** [comment]
1.) On my jwsteeb AWS account, I can see public buckets but when I try: https://us-east-1.console.aws.amazon.com/s3/buckets/viper-test-data?region=us-east-1&bucketType=general&tab=objects it says Insufficient permissions to list objects.

2.) In read_processing_set I removed the try except PermissionError and s3 = s3fs.S3FileSystem(anon=False, requester_pays=False) works. This also seems to indicate that the bucket is not public.

3.) In test_s3_read_processing_set why is "s3://viper-test-data/Antennae_North.cal.lsrk.split.ms" using .ms instead of .vis.zarr?

**Comment by Jan-Willem** [comment]
I have also merged the latest version of main into this branch. This has no impact on the code of this pull request.

**Comment by amcnicho** [comment]
Adding a GetObject policy to all objects in the zarr dataset bucket was also necessary in order to avoid Permission and NoCredential Errors from s3fs/boto. After additional refactoring to force the use of `signature_version=UNSIGNED` (for the `anon=True` case), the new test of anonymous public reads passes CI tests, and local testing with AWS credentials available still work as expected. Unless there are other tests desired, this seems ready :shipit:

File: src/xradio/vis/read_processing_set.py [hunk 1]
```diff
@@ -18,44 +20,101 @@ def read_processing_set(
...
+
+    elif ps_store.startswith("s3"):
+        # only if not found locally, check if dealing with an S3 bucket URL
+        ps_store_is_s3dir = True
+        if not ps_store.endswith("/"):
+            # just for consistency, as there is no os.path equivalent in s3fs
+            ps_store = ps_store + "/"
+
+        try:
+            # initiatlize the S3 "file system", first attempting to use pre-configured credentials
```
Comment (vsuorant) [inline src/xradio/vis/read_processing_set.py:46]: Typo

Comment (amcnicho) [reply src/xradio/vis/read_processing_set.py]: Fixed

**Review by vsuorant — APPROVED** [review]

**Pull request merged**